### PR TITLE
feat: api videos authentication

### PR DESCRIPTION
**Pull request by mikeallisonJS**
# Description

### Issue

videos admin frontend needs specific authentication.

### Solution

Add authentication & query to api-videos

### CODE REVIEW

File: apps/api-videos/db/schema.prisma [hunk 1]
```diff
@@ -256,3 +256,17 @@ model Keyword {
...
+  youtubeAdmin
+  coreMember
+  coreAdmin
+}
+
+model VideoAdminUser {
+  id     String               @id @default(uuid())
+  userId String               @unique
+  roles  VideoAdminUserRole[]
+}
```
Comment (tataihono) [inline apps/api-videos/db/schema.prisma:272]:
```suggestion
model VideoUserRole {
  id     String               @id @default(uuid())
  userId String               @unique
  roles  VideoRole[]
}
```

File: apps/api-videos/db/schema.prisma [hunk 2]
```diff
@@ -256,3 +256,17 @@ model Keyword {
...
   @@index([value(type: Hash)])
 }
+
+enum VideoAdminUserRole {
+  admin
+  youtubeMember
+  youtubeAdmin
+  coreMember
+  coreAdmin
+}
```
Comment (tataihono) [inline apps/api-videos/db/schema.prisma:266]: lets start simple. Everyone can manage everything for now.

```suggestion
enum VideoRole {
  publisher
}
```

File: apps/api-videos/db/schema.prisma (hunk 1 above)
Comment (tataihono) [reply apps/api-videos/db/schema.prisma:272]: more in line with the journeys labelling of this table.

File: apps/api-videos/src/schema/video/video.ts [hunk 3]
```diff
@@ -257,5 +258,14 @@ builder.queryFields((t) => ({
...
+  }),
+  currentVideoRoles: t.field({
+    type: [VideoAdminUserRole],
+    nullable: false,
+    authScopes: {
+      isAuthenticated: true
+    },
+    resolve: async (_query, _args, ctx: Context) =>
+      ctx.currentVideoUser?.roles ?? []
   })
```
Comment (tataihono) [inline apps/api-videos/src/schema/video/video.ts:270]: can we break this out into a user file and just extend the user object out? that way the me call could be:
```
me {
  id
  videoUserRoles {
    id
    roles
  }
}

File: apps/api-videos/src/yoga.ts [hunk 4]
```diff
@@ -4,13 +4,32 @@ import {
...
+      token == null || token === '' ? null : await getUserFromAuthToken(token)
+
+    return {
+      currentUser,
+      currentVideoUser:
+        currentUser != null
+          ? await prisma.videoAdminUser.findUnique({
+              where: { userId: currentUser.id }
+            })
+          : null,
```
Comment (tataihono) [inline apps/api-videos/src/yoga.ts:29]:
```suggestion
      currentRoles:
        currentUser != null
          ? await prisma.videoAdminUser.findUnique({
              where: { userId: currentUser.id }
            })?.roles
          : null,
```

File: libs/yoga/jest.config.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import type { Config } from 'jest'
+
+const config: Config = {
+  displayName: 'nest-decorators',
```
Comment (tataihono) [inline libs/yoga/jest.config.ts:4]:
```suggestion
  displayName: 'yoga',
```

File: libs/yoga/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# nest-decorators
+
+This library was generated with [Nx](https://nx.dev).
+
+## Running unit tests
+
+Run `nx test nest-decorators` to execute the unit tests via [Jest](https://jestjs.io).
```
Comment (tataihono) [inline libs/yoga/README.md:7]:
```suggestion
Run `nx test yoga` to execute the unit tests via [Jest](https://jestjs.io).
```

File: libs/yoga/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# nest-decorators
```
Comment (tataihono) [inline libs/yoga/README.md:1]:
```suggestion
# yoga
```

File: libs/yoga/project.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  "targets": {
+    "lint": {
+      "executor": "@nx/eslint:lint",
+      "outputs": ["{options.outputFile}"]
+    },
+    "test": {
+      "executor": "@nx/jest:jest",
+      "outputs": ["{workspaceRoot}/coverage/libs/yoga"],
+      "options": {
+        "jestConfig": "libs/nest/decorators/jest.config.ts"
```
Comment (tataihono) [inline libs/yoga/project.json:15]:
```suggestion
        "jestConfig": "libs/yoga/jest.config.ts"
```

File: libs/yoga/project.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      }
+    },
+    "codecov": {
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "codecov -f coverage/libs/yoga/cobertura-coverage.xml -F libs.nest.decorators"
```
Comment (tataihono) [inline libs/yoga/project.json:21]:
```suggestion
        "command": "codecov -f coverage/libs/yoga/cobertura-coverage.xml -F libs.yoga"
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-media/src/workers/algolia/service/service.ts [hunk 10]
```diff
@@ -43,7 +43,16 @@ interface LanguageRecord {
 
 async function getLanguages(logger?: Logger): Promise<LanguageRecord> {
   try {
-    const { data } = await apollo.query({
+    const { data } = await apollo.query<{
```
[the remark this answers is not in the document]
Comment (mikeallisonJS) [reply apps/api-media/src/workers/algolia/service/service.ts]: fixed with multi-schema

File: apps/api-media/src/workers/bigQuery/importers/languageSlugs/languageSlugs.ts [hunk 11]
```diff
@@ -36,7 +36,9 @@ export async function importLanguageSlugs(
 ): Promise<() => void> {
   logger?.info('fetch language slugs from api-languages started')
 
-  const { data } = await apollo.query({
+  const { data } = await apollo.query<{
+    languages: Array<{ id: string; slug: string | null }>
+  }>({
```
[the remark this answers is not in the document]
Comment (mikeallisonJS) [reply apps/api-media/src/workers/bigQuery/importers/languageSlugs/languageSlugs.ts]: fixed by multi-schema

File: apps/api-media/src/schema/user/index.ts [hunk 12]
```diff
@@ -0,0 +1 @@
+import './user'
```
Comment (tataihono) [inline apps/api-media/src/schema/user/index.ts:1]:
```suggestion
import './enums'
import './user'
```

File: apps/api-media/src/schema/user/user.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+      authScopes: {
+        isAuthenticated: true
+      },
+      resolve: async (data) => {
+        return (
+          (
+            await prisma.userMediaRole.findUnique({
+              where: { id: data.id }
+            })
+          )?.roles ?? []
```
Comment (tataihono) [inline apps/api-media/src/schema/user/user.ts:22]: can we do a find or create here?

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-media/src/schema/user/user.ts (hunk 13 above)
Comment (mikeallisonJS) [reply apps/api-media/src/schema/user/user.ts:22]: will implement later if needed

**Review by tataihono — APPROVED** [review]

**Pull request merged**